.github/workflows/release-integration.yml

```diff
@@ -58,34 +75,61 @@ jobs:
       DOCKER_PLATFORMS: "linux/amd64,linux/arm64,linux/arm" # Must be consistent with the matrix from the job above.
     steps:
       - uses: actions/checkout@v2
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/download-artifact@v2
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.FSI_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.FSI_DOCKERHUB_TOKEN }}
 
       - name: Generate Docker image version from git tag
         run: |
           echo "${{ github.event.release.tag_name }}" | grep -E '^v[0-9.]*[0-9]$'
           DOCKER_IMAGE_TAG=$(echo "${{ github.event.release.tag_name }}" | sed 's/^v//')
           echo "DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG" >> $GITHUB_ENV
-      - if: ${{ github.event.release.prerelease }}
+
+      - name: Change Docker image version if this run is a pre-release
+        if: ${{ github.event.release.prerelease }}
         run: |
           echo "DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG}-pre" >> $GITHUB_ENV
-      - uses: docker/setup-qemu-action@v1
-      - uses: docker/setup-buildx-action@v1
-
-      - name: Download all artifacts from build job
-        uses: actions/download-artifact@v2
-
-      - uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.FSI_DOCKERHUB_USERNAME }}
-          password: ${{ secrets.FSI_DOCKERHUB_TOKEN }}
 
       - name: Build and push Docker image
+        if: ${{ github.event.release.prerelease }}
```

Comment:
why pushing the image only when pre-releaseing?

Reply:
oh i see the change, I think the previous logic was more easy to read. One step for pushing  the versioned tag and other the latest.
Is there any reason for this change?

Reply:
Solved in [#2025837](https://github.com/newrelic/newrelic-k8s-metrics-adapter/pull/40/commits/2025837431e05ff2137e0c2929b3842de0f5ea7e)

Reply:
I do not have any preference 😄

Reply:
> But in case we need other pipelines to install something there is a chance that the layer created misses the cache and the image with the latest is different than the one with a vX.Y.Z.

oh good catch ! thanks for clarifying it